### ISSUE TITLE
pouchdb-hoodie-api: use .changes() for events

### ISSUE THREAD

**Issue by gr2m** [hoodiehq/camp#13]
### 🤔 What you will need to know

**This one is challenging**. I tried it myself a few months ago and failed. You should know the JavaScript Event Loop, understand why an immediately resolved Promise and `process.nextTick()` do not call their callbacks always in the same order, ideally find a way to change that or a work around.

### :dart: The goal

We once had a very simple, beautiful solution for events in [pouchdb-hoodie-api](https://github.com/hoodiehq/pouchdb-hoodie-api): we started listening to PouchDB’s changes feed and used that to emit `add`, `update` and `change` events. But then we hit https://github.com/hoodiehq/pouchdb-hoodie-api/issues/54 which caused a timing issue, and after trying for a while to work around the problem while still using the changes feed, we eventually gave up. Instead of listening to the changes feed, we wrap all our methods and emit events manually, see changes here: https://github.com/hoodiehq/pouchdb-hoodie-api/pull/56/files

Another problem with just using `.changes()` is that we cannot differentiate between changes coming from remote (via sync) or changes created locally. 

But we feel there is a better way, and it’s worth looking for it, it would remove a lot of complexity. And whoever finds a solution for this, gets a special Hoodie treat that I’ll have to make up first, but I promise it will be something nice <3

### :clipboard: Step by Step

- [x] 🙋 **Claim this issue**: Comment below
  Please only claim if you want to start working on it during [the event](https://www.meetup.com/WriteSpeakCode-SFBay/events/235091424/).
  Once claimed we add you as contributor to this repository.
- [x] 👌 **Accept our invitation** to this repository. Once accepted, assign yourself to this repository
- [x] 👓 **Please review** our [Code of Conduct](http://hood.ie/code-of-conduct/)  
  In a nutshell: be patient and actively kind

- [x] **[Setup the hoodie repository locally](https://github.com/hoodiehq/hoodie-account-client#testing)** and make sure that all tests pass
- [ ] **Find a solution** 💚  
- [ ] **Start a Pull Request**. Mention `closes hoodiehq/camp#13` in the description.  
  You don’t need to be finished to start the PR, I’d suggest to start it as soon as possible so we can discuss different approaches and collaborate on it
- [ ] **Done** :+1: Replace the `in progress` label with `ready`. Ask in comments for a review :)

Ping us in the [Hoodie Chat](http://hood.ie/chat/) or on [Twitter](https://twitter.com/hoodiehq/) if you have any questions :)

**Comment by jcquinlan** [comment]
Gonna take a stab at it, but no guarantees, haha. Forking now - I'll keep you in the (event) loop.

**Comment by gr2m** [comment]
good luck @jcquinlan 👍

**Comment by dweremeichik** [comment]
@gr2m @jcquinlan  `Start a Pull Request. Mention closes hoodiehq/camp#11 in the description.` Should say 13, not 11. :+1:

**Comment by gr2m** [comment]
@dweremeichik fixed, thanks a lot :)

**Comment by gr2m** [comment]
@jcquinlan I’ve invited you to this repository, you can accept the invitation at https://github.com/hoodiehq/camp/invitations. Once accepted, you can assign yourself and start checking off the check boxes above

**Comment by gr2m** [comment]
I’ll free up this issue for [Write/Speak/Code Open Source Hours](https://www.meetup.com/WriteSpeakCode-SFBay/events/235091424/) as it was not worked on in the past month

**Comment by gr2m** [comment]
I finally know how we can use the `.changes()` feed but still differentiate between local and remote docs, so part of the problem here can be resolved now.

The trick is to generate a local client uuid once and then persist it, and store that client uuid with every document update.

Now when we listen to changes of the localDatabase, we can simply check if the clientId is the one of the current client or not. Dependent on that we pass `remote: true` or `remote: false`.

I’ll give this another try and see if I can resolve the timing issue as well.

**Comment by gr2m** [comment]
done via https://github.com/hoodiehq/pouchdb-hoodie-api/issues/123
